File: backend/src/PortalRRHHFZ.Domain/Entities/Alerta.cs
using PortalRRHHFZ.Domain.Enums;

namespace PortalRRHHFZ.Domain.Entities;

public sealed class Alerta : AuditableEntity
{
    public int AlertaId { get; set; }
    public TipoAlerta TipoAlerta { get; set; }
    public EstadoAlerta EstadoAlerta { get; set; } = EstadoAlerta.Pendiente;
    public int ColaboradorId { get; set; }
    public int? DocumentoColaboradorId { get; set; }
    public DateTime FechaVencimiento { get; set; }
    public string Mensaje { get; set; } = string.Empty;
    public DateTime FechaGeneracion { get; set; }
    public DateTime? FechaGestion { get; set; }
    public int? GestionadaPor { get; set; }
    public string? ObservacionGestion { get; set; }

    public Colaborador Colaborador { get; set; } = null!;
    public DocumentoColaborador? DocumentoColaborador { get; set; }
    public Usuario? UsuarioGestion { get; set; }
}

File: frontend/src/types/alerta.ts
export type EstadoAlerta = 'Pendiente' | 'Gestionada' | 'Vencida' | 'Ignorada';

export type TipoAlerta =
  | 'Cedula'
  | 'Licencia'
  | 'Contrato'
  | 'PeriodoProbatorio'
  | 'Documento';

export type AlertaFilters = {
  estadoAlerta?: EstadoAlerta;
  tipoAlerta?: TipoAlerta;
  colaboradorId?: number;
  desde?: string;
  hasta?: string;
  incluirInactivas?: boolean;
};

export type AlertaFilterValues = {
  estadoAlerta: string;
  tipoAlerta: string;
  colaboradorId: string;
  desde: string;
  hasta: string;
  incluirInactivas: boolean;
};

export type AlertaList = {
  alertaId: number;
  tipoAlerta: string;
  estadoAlerta: string;
  colaboradorId: number;
  nombreCompletoColaborador: string;
  documentoColaboradorId: number | null;
  tipoDocumentoNombre: string | null;
  fechaVencimiento: string;
  mensaje: string;
  fechaGeneracion: string;
  fechaGestion: string | null;
  gestionadaPor: number | null;
  gestionadaPorNombre: string | null;
  observacionGestion: string | null;
  isActive: boolean;
};

export type AlertaResumen = {
  totalAlertas: number;
  pendientes: number;
  vencidas: number;
  gestionadas: number;
  ignoradas: number;
  porTipoAlerta: AlertaPorTipo[];
  proximasAVencer: number;
  vencidasPendientes: number;
};

export type AlertaPorTipo = {
  tipoAlerta: string;
  total: number;
};

export type GestionarAlertaRequest = {
  observacionGestion: string | null;
};

export type RecalcularAlertasResult = {
  alertasCreadas: number;
  alertasActualizadasAVencidas: number;
  totalAlertasActivas: number;
};
